File: 2021/1/solution.cpp
#include <fstream>
#include <iostream>
using namespace std;

void part_one()
{
    ifstream file("input");
    string line;
    int count = 0;
    int last = -1;
    getline(file, line);
    while (!line.empty()) {
        int n = stoi(line);
        if (n > last)
            count++;
        last = n;
        getline(file, line);
    }
    cout << count << '\n';
}

void part_two()
{
    ifstream file("input");
    string line;
    int count = 0;
    int last[3] = { -1, -1, -1 };
    for (int i = 0; i < 3; i++) {
        getline(file, line);
        last[i] = stoi(line);
    }
    while (!line.empty()) {
        int n = stoi(line);
        if (n > last[0])
            count++;
        for (int i = 0; i < 2; i++)
            last[i] = last[i + 1];
        last[2] = n;
        getline(file, line);
    }
    cout << count << '\n';
}

int main()
{
    part_one();
    part_two();
    return 0;
}

File: 2021/1/solution.py

def part_one():
    file = open("input", "r")
    ints = list(map(int, file.readlines()))
    j = -1
    count = 0
    for i in ints:
        if i > j:
            count +=1
        j = i
    print(count)

def part_two():
    file = open("input", "r")
    ints = list(map(int, file.readlines()))
    count = 0
    window = ints[:3]
    for i in ints[3:]:
        if i > window.pop(0):
            count+=1
        window.append(i)
    print(count)

part_two()
